logstash-core/lib/logstash/agent_plugin_registry.rb

```diff
@@ -0,0 +1,66 @@
+# encoding: utf-8
+require "logstash/agent"
+
+module LogStash module AgentPluginRegistry
```

Comment:
As we spoke, I wonder why the registry should be a module. It have some state, as seen in the registry lookup. Can you walk me through you thinking?